### PR TITLE
Plugin name search and serial devices port address changes

### PR DESCRIPTION
**Pull request by TiffanyLuce**
Sorry I was stupid enough to make two different changes to the same branch.

   * First thing is for plugins ending with numbers like caen_n1470 ; if you have two different devices and you want to call them caen_n14701 and 2, they won't get recognized because doberman just deletes all numbers at the end. Now there is the possibility to call your devices caen_n1470_1 and it will ignore the last underscore and everything that's after (if it didn't find anything by the name and the name - the last numbers).
   * Second is for serial devices to be searched by id rather than by ttyUSBX. Because this tty address changes everytime you unplug your device or restart your computer while the id should not change.

Both changes have been tested for this caen module and it works for them.

### CODE REVIEW

File: Doberman/utils.py [hunk 1]
```diff
@@ -29,16 +29,22 @@ def find_plugin(name, path):
...
+    split = False
     if not isinstance(path, (list, tuple)):
         path = [path]
     spec = importlib.machinery.PathFinder.find_spec(name, path)
     if spec is None:
         strip = True
         spec = importlib.machinery.PathFinder.find_spec(name.strip('0123456789'), path)
+    if spec is None:
+        split = True
+        spec = importlib.machinery.PathFinder.find_spec('_'.join(name.split('_')[:-1]), path)
```
Comment (adambrown1) [inline Doberman/utils.py:41]: rsplit cleaner?

File: Doberman/utils.py [hunk 2]
```diff
@@ -29,16 +29,22 @@ def find_plugin(name, path):
     :returns constructor: the constructor of the requested device
     """
     strip = False
+    split = False
```
Comment (adambrown1) [inline Doberman/utils.py:32]: This has got a bit messy with all the binary variables and then ifs below. Maybe make one variable be the plugin name and then save that the first time you got through the possibilities, to get rid of the second if?

File: Doberman/BaseDevice.py [hunk 3]
```diff
@@ -207,16 +207,23 @@ def setup(self):
...
-        try:
-            if self.tty.startswith('/'):  # Full path to device TTY specified
-                self._device.port = self.tty
-            else:
-                self._device.port = f'/dev/tty{self.tty}'
-            self._device.open()
-        except serial.SerialException as e:
-            raise ValueError(f'Problem opening {self._device.port}: {e}')
+        else:
+            try:
```
Comment (adambrown1) [inline Doberman/BaseDevice.py:219]: Can you move the try outside the if to combine both exception handling parts (since they are the same)

**Review by adambrown1 — APPROVED** [review]
Only a few stylistic suggestions

File: Doberman/utils.py (hunk 1 above)
Comment (doberman-freiburg) [reply Doberman/utils.py:41]: ohhhh, I haven't seen that function! indeed, will change, thanks!

**Comment by TiffanyLuce** [comment]
I should have fixed the few suggestions + changed the base branch to experimental under Jaron's suggestion to then merge experimental all at once in master

**Review by adambrown1 — APPROVED** [review]
Beautiful

**Pull request merged**